ui/hooks/useCurrencyRatePolling.test.ts

```diff
@@ -0,0 +1,119 @@
+import { renderHookWithProvider } from '../../test/lib/render-helpers';
+import {
+  currencyRateStartPolling,
+  currencyRateStopPollingByPollingToken,
+} from '../store/actions';
+import useCurrencyRatePolling from './useCurrencyRatePolling';
+
+let mockPromises: Promise<string>[];
+
+jest.mock('../store/actions', () => ({
+  currencyRateStartPolling: jest.fn().mockImplementation((input) => {
+    const promise = Promise.resolve(`${input}_rates`);
+    mockPromises.push(promise);
+    return promise;
+  }),
+  currencyRateStopPollingByPollingToken: jest.fn(),
+}));
+
+describe('useCurrencyRatePolling', () => {
+  beforeEach(() => {
+    mockPromises = [];
+    jest.clearAllMocks();
+  });
+
+  it('should poll currency rates for native currencies when enabled and stop on dismount', async () => {
+    const state = {
+      metamask: {
+        isUnlocked: true,
+        completedOnboarding: true,
+        useCurrencyRateCheck: true,
+        selectedNetworkClientId: 'selectedNetworkClientId',
+        networkConfigurationsByChainId: {
```

Comment:
Can we add two networks here and ensure it's called twice?

Reply:
done [5c5b246](https://github.com/MetaMask/metamask-extension/pull/28646/commits/5c5b2468aeda13870ab2db21b6b08ae12e6aea57)